test/unittest/unittest_nntrainer_tensor.cpp

```diff
@@ -4460,6 +4460,69 @@ TEST(nntrainer_Tensor, dequantize_05_n) {
   EXPECT_THROW({ input.dequantize(output, 1); }, std::invalid_argument);
 }
 
+TEST(nntrainer_neon_experimental, trigonometric_simd) {
+  int batch = 1;
+  int channel = 1;
+  int height = 1440;
+  int width = 1440;
+
+  const int MOD = 10;
+
+  const float eps = 1e-6;
+
+  nntrainer::Tensor input(batch, channel, height, width);
+  nntrainer::Tensor sin_output(batch, channel, height, width);
+  nntrainer::Tensor cos_output(batch, channel, height, width);
+  GEN_TEST_INPUT(input, (i * (channel * width * height) + j * (height * width) +
+                         k * (width) + l + 1) %
+                          MOD);
+
+  nntrainer::Tensor result_sine(batch, channel, height, width);
+  nntrainer::Tensor result_cosine(batch, channel, height, width);
+
+  for (int b = 0; b < batch; b++) {
+    for (int c = 0; c < channel; c++) {
+      for (int h = 0; h < height; h++) {
+        for (int w = 0; w < width; w++) {
+          result_sine.setValue(b, c, h, w,
+                               std::sin(input.getValue(b, c, h, w)));
+          result_cosine.setValue(b, c, h, w,
+                                 std::cos(input.getValue(b, c, h, w)));
+        }
+      }
+    }
+  }
+
+  input.sin_transform(sin_output);
+  input.cos_transform(cos_output);
+
+  bool flag = true;
+
+  for (int b = 0; b < batch; b++) {
+    for (int c = 0; c < channel; c++) {
+      for (int h = 0; h < height; h++) {
+        for (int w = 0; w < width; w++) {
+          double sin_err = std::abs(sin_output.getValue(b, c, h, w) -
+                                    result_sine.getValue(b, c, h, w));
+          double cos_err = std::abs(cos_output.getValue(b, c, h, w) -
+                                    result_cosine.getValue(b, c, h, w));
+          if (sin_err > eps || cos_err > eps)
+            flag = false;
+
+          if (sin_err > eps)
+            std::cout << sin_output.getValue(b, c, h, w) << " VS "
+                      << result_sine.getValue(b, c, h, w) << std::endl;
+          if (cos_err > eps)
+            std::cout << cos_output.getValue(b, c, h, w) << " VS "
+                      << result_cosine.getValue(b, c, h, w) << std::endl;
+        }
+      }
+    }
+  }
+
+  EXPECT_EQ(flag, true);
```

Comment:
```suggestion
  EXPECT_EQ(sin_output, result_sine);
  EXPECT_EQ(cos_output, result_cosine);
```

quick question, any reason for manually checking each value instead of comparing two tensors?

Reply:
Since we are relying on `neon_mathfunc.h` for trigonometric computation, we need an epsilon value to assess